src/App.js

```diff
@@ -40,8 +41,17 @@ class App extends Component {
 
   componentDidMount() {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
-    .then((response) => response.json())
+    .then((response) => {
+      if(!response.ok) {
+        this.setState({error: 'server error'})
+        throw Error(response.status)
```

Comment:
Consider moving the 'throw Error' into the API call within Apicall.js to adhere to the DRY principle. Please let me know if you have any questions about this. In this context, you can use the catch block to handle any errors that are returned from the API.